src/datastores/handlers/base.js

```diff
@@ -147,7 +147,7 @@ class Playlists {
   static deleteVideoIdsByPlaylistId(_id, videoIds) {
     return db.playlists.updateAsync(
       { _id },
-      { $pull: { videos: { $in: videoIds } } },
+      { $pull: { videos: { videoId: { $in: videoIds } } } },
```

Comment:
Is this a fix?

Reply:
Yes, it is. But this function is not in use in your PR, it seems. You can add this fix as well.